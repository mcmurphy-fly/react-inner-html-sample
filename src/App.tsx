import React from 'react';
import InnerContent from './components/InnerContent';

const App = () => (
  <>
    <InnerContent
      content={`
        <h1>Hello</h1>
        <p>bla bla <b>bla</b></p>
    `}
      tagName="section"
    />
    <InnerContent content="<p>abc<iframe//src=jAva&Tab;script:alert(3)>def</p>" />
  </>
);

export default App;
