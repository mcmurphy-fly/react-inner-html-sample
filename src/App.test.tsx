import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

describe('App', () => {
  test('matches snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
