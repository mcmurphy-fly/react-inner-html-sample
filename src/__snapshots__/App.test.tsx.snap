// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
Array [
  <section
    dangerouslySetInnerHTML={
      Object {
        "__html": "
        <h1>Hello</h1>
        <p>bla bla <b>bla</b></p>
    ",
      }
    }
  />,
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>abc</p>",
      }
    }
  />,
]
`;
